### PR TITLE
housekeeping: grpc-gateway v2 upgrade

### PR DESCRIPTION
**Pull request by danielhochman**
### Description
Upgrade to [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) v2.

Changes were fairly minor other than a couple of upstream bugfixes.

It did require a change to the `go_package` directive in each proto, since the latest protoc-gen-go emits the following:
```
2020/10/16 17:09:53 WARNING: Deprecated use of 'go_package' option without a full import path in "resolver/v1/resolver_api.proto", please specify
```

### Testing Performed
Manual, CI.

### CODE REVIEW

**Review by mcutalo88 — APPROVED** [review]
Looks good I checked most of the `go_package` path changes and they seemed good. Left one question just to clarify my understading, but nothing blocking.

File: tools/compile-protos.sh [hunk 1]
```diff
@@ -190,13 +176,6 @@ discover_protos() {
   done <  <(find "${API_ROOT}" -name '*.proto' -exec dirname {} \; | tr '\n' '\0' | sort -sdzu)
 }
 
-discover_core_protos() {
```
[the remark this answers is not in the document]
Comment (danielhochman) [reply tools/compile-protos.sh:193]: these were originally mappings for the proto filename to the full package path. now that there's a full package path in go_package it's not necessary.

**Comment by danielhochman** [comment]
@mcutalo88 oh i should note that i wrote a script to generate the new go_package values, so they should be good.

File: tools/compile-protos.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcutalo88) [reply tools/compile-protos.sh:193]: Ah sweet thats awesome thanks

File: tools/compile-protos.sh [hunk 2]
```diff
@@ -124,36 +123,23 @@ main() {
   proto_out_dir="${REPO_ROOT}/backend/api"
   mkdir -p "${proto_out_dir}"
 
-  MFLAGS=""
```
Comment (dschaller) [inline tools/compile-protos.sh:127]: 🎉

**Pull request merged**